### DISCUSSION TITLE
passing parameters into rshell scripts

### DISCUSSION THREAD

**Discussion by KSt-HW** [micropython/micropython#9165] [General]
Hello everyone, i am new here, so a big "Hi!" to all of you. Maybe you can even help  a noob like me.

I am using rshell to manipulate my memory on a device running MicroPython. Since i often have to do the same things over and over again, i like to work with scripts like this:
`rshell.exe -p <serieal_port> -f script.rsh` 
(yes, i am working with WIN)

Now i have the need to pass some parameter into the script. Is that possible?
I tried different things, but none worked, (e.g.
`rshell.exe -p <serieal_port> -f script.rsh <parameter>`
`rshell.exe -p <serieal_port> -f 'script.rsh <parameter>'`
`rshell.exe -p <serieal_port> -f "script.rsh <parameter>"`
). I also searched the web but didn't find anything helpful neither. So i wonder, if that is even a thing?

Thanks a lot for any help or explanation!

**Comment by robert-hh** [comment]
Parameters to windows batch scripts are denoted as %1 %2 %3 ....

**Reply by KSt-HW** [reply]
Yes, i know and use that successfully. Thanks!

**Comment by peterhinch** [comment]
I don't know of a way, but perhaps @dhylands will comment.

I have a [fork](https://github.com/peterhinch/rshell) of `rshell` which provides for text macros at the `>` prompt. These can take args and simplify managing complex projects.

**Reply by KSt-HW** [reply]
Tanks a lot! I will have a look at it.

**Comment by CroftAr** [comment]
When you launch rshell.exe you pass it the command line, your environment, and a script of commands.  The rshell docs don't mention using the environment or handling parameters, so you're left with modifying the script or, as Peter suggests, how its commands work.  The first approach that comes to mind is:
  sed 's/parameter/value/g' script_template.rsh > script.rsh ;   rshell.exe -p <serieal_port> -f script.rsh
or create a parameterised rshell_macros.py for the rshell that supports macros.

**Reply by KSt-HW** [reply]
Thanks a lot for your answer! So i will have to look into using macros, hmkay.
